src/v/kafka/server/requests.cc

```diff
@@ -249,6 +249,14 @@ process_result_stages process_request(
   request_context&& ctx,
   ss::smp_service_group g,
   const session_resources& sres) {
+    auto& key = ctx.header().key;
+
+    if (
+      unlikely(ctx.sasl() && ctx.sasl()->complete() && ctx.sasl()->expired())
```

Comment:
seems like the part of this condition which would be unlikely is that the client isn't well behaved, which would seem to be that a client let it's connection expire?

in any case, you could probably also remove `unlikely` under the argument that redpanda can't make a static decision about knowing what a client configuration / behavior is going to be.